core/primitives/block/eligible.hpp

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "cbor_blake/ipld_version.hpp"
+#include "primitives/tipset/tipset.hpp"
+#include "vm/actor/builtin/states/miner/miner_actor_state.hpp"
+#include "vm/actor/builtin/states/storage_power/storage_power_actor_state.hpp"
+#include "vm/actor/builtin/types/storage_power/policy.hpp"
+#include "vm/state/impl/state_tree_impl.hpp"
+
+namespace fc::miner_eligible_to_miner {
```

Comment:
I think the right place for file is in `core/blockchain/block_validator` and namespace 'blockchain::block_validator`.